src/ScottPlot5/ScottPlot5/Plot.cs

```diff
@@ -28,6 +28,8 @@ public class Plot : IDisposable
     public IBenchmark Benchmark { get; set; } = new StandardBenchmark();
     public IZoomRectangle ZoomRectangle { get; set; }
     internal RenderDetails LastRenderInfo { get; set; } = new();
+
+    public PixelRect FigureRect => LastRenderInfo.FigureRect;
```

Comment:
I'd really like to avoid exposing outdated state like this...

Can we change this to something that makes it really intentional what is being returned, maybe like this?

```cs
public RenderDetails GetLastRenderInfo() => LastRenderInfo;
```

Or I guess we could just make `LastRenderInfo` `public` instead of `internal`

In ScottPlot 5 the plot should store no pixel state. The figure size is state that is passed into the render method, and ideally it never leaves that stack.

Reply:
I do like keeping `LastRenderInfo` internal, most of its members shouldn't be available publicly and it's a struct that we could well want to modify in the future.

That said, I think we need to be able to know how big the image we create is, though that doesn't necessarily need to live in `Plot`. One option is to create a `PlotSnapshot` class, which holds the image dimensions, and something like an `SKImage`. This class can then implement `SavePng`, `SaveJpeg`, etc etc.

This way we're very explicit that we're only sharing the dimensions as they were when the snapshot was created, and we also make it a lot easier for someone to create a class that exports to their own funky formats, especially those that we're unlikely to support (e.g. 4.1 supported TIFF and BMP, which don't currently work in 5.0, plus I could see people wanting PDF and SVG)

Reply:
> we need to be able to know how big the image we create is

`Render()` is `void` but maybe it should return `RenderDetails` 🤔

Reply:
I think this is resolved (now that we agreed not to return `RenderDetails`)